src/sig_stfl/sig_stfl.h

```diff
@@ -0,0 +1,578 @@
+	void *secret_key_data;
+
+	/* mutual exclusion struct */
+	void *mutex;
+
+	/* Application managed data related to secure storage of secret key data */
+	void *context;
+
+	/**
+	 * Secret Key retrieval Function
+	 *
+	 * @param[in] sk The secret key represented as OQS_SIG_STFL_SECRET_KEY object
+	 * @param[out] sk_len length of the private key as a byte stream
+	 * @param[out] sk_buf_ptr pointer to private key data as a byte stream
+	 * @returns length of key material data available
+	 * Caller is responsible for **deallocating** the pointer to buffer `sk_buf_ptr`.
+	 */
+	OQS_STATUS (*serialize_key)(uint8_t **sk_buf_ptr, size_t *sk_len, const OQS_SIG_STFL_SECRET_KEY *sk);
+
+	/**
+	 * Secret Key to internal structure Function
+	 *
+	 * @param[in] sk OQS_SIG_STFL_SECRET_KEY object
+	 * @param[in] key_len length of the returned byte string
+	 * @param[in] sk_buf The secret key data to populate the key object
+	 * @param[in] context application-specific data
+	 *            used to keep track of this secret key stored in a secure manner.
+	 *            The application manages this memory.
+	 * @returns  status of the operation populated with key material none zero length.
+	 * Caller is responsible to **unallocate** the buffer `sk_buf`.
+	 */
+	OQS_STATUS (*deserialize_key)(OQS_SIG_STFL_SECRET_KEY *sk, const size_t sk_len, const uint8_t *sk_buf, void *context);
+
+	/**
+	 * Secret Key Locking Function
+	 *
+	 * @param[in] mutex application defined mutex
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*lock_key)(void *mutex);
+
+	/**
+	 * Secret Key Unlocking / Releasing Function
+	 *
+	 * @param[in]  mutex application defined mutex
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*unlock_key)(void *mutex);
+
+	/**
+	 * Store Secret Key Function
+	 * Callback function used to securely store key data after a signature generation.
+	 * When populated, this pointer points to the application supplied secure storage function.
+	 * @param[in] sk_buf The serialized secret key data to secure store
+	 * @param[in] buf_len length of data to secure
+	 * @param[in] context application supplied data used to locate where this secret key
+	 *            is stored (passed in at the time the function pointer was set).
+	 *
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 * Ideally written to secure device
+	 */
+	OQS_STATUS (*secure_store_scrt_key)(uint8_t *sk_buf, size_t buf_len, void *context);
+
+	/**
+	 * Free internal variant-specific data
+	 *
+	 * @param[in] sk The secret key represented as OQS_SIG_STFL_SECRET_KEY object
+	 * @return none
+	 */
+	void (*free_key)(OQS_SIG_STFL_SECRET_KEY *sk);
+
+	/*
+	 * Secure storage for private keys used in stateful signature schemes is outside the scope of the OQS library.
+	 * This is the responsibility of any adopting application. The application must supply
+	 * a function to for this purpose. A callback function and context data must be set in-order
```

Comment:
typos? "to for"? "in-order"?

Reply:
Fixed typo in https://github.com/open-quantum-safe/liboqs/pull/1655